ddtrace/contrib/sanic/patch.py

```diff
@@ -18,34 +19,37 @@
 
 config._add("sanic", dict(_default_service="sanic", distributed_tracing=True))
 
+SANIC_PRE_21 = None
+
+
+def update_span(span, response):
+    if isinstance(response, sanic.response.BaseHTTPResponse):
+        status_code = response.status
+        response_headers = response.headers
+    else:
+        # invalid response causes ServerError exception which must be handled
+        status_code = 500
+        response_headers = None
+    trace_utils.set_http_meta(span, config.sanic, status_code=status_code, response_headers=response_headers)
+
 
 def _wrap_response_callback(span, callback):
-    # wrap response callbacks (either sync or async function) to set span tags
-    # based on response and finish span before returning response
-
-    def update_span(response):
-        if isinstance(response, sanic.response.BaseHTTPResponse):
-            status_code = response.status
-            response_headers = response.headers
-        else:
-            # invalid response causes ServerError exception which must be handled
-            status_code = 500
-            response_headers = None
-        trace_utils.set_http_meta(span, config.sanic, status_code=status_code, response_headers=response_headers)
-        span.finish()
```

Comment:
Was it intentional to leave out `span.finish()` when extracting this `update_span()` out to top-level? Just wanted to clarify if this was due to the change in `patch_handle_request()` of using context managers for creating spans.

Reply:
It is indeed due to the `Sanic.handle_request` signature change. As explained in the PR description, the request span is now finished when the `Sanic.handle_request` returns instead of when `write_callback` or `stream_callback` is called.